cmd/ndm_daemonset/controller/diskstore.go

```diff
@@ -162,14 +162,13 @@ func (c *Controller) GetExistingDiskResource(listDr *apis.DiskList, uuid string)
 // list of active resources. One active resource which is present in etcd not in
 // system that will be marked as inactive.
 func (c *Controller) DeactivateStaleDiskResource(devices []string) {
-	listDisks := append(devices, GetActiveSparseDisksUuids(c.HostName)...)
 	listDR, err := c.ListDiskResource()
 	if err != nil {
 		glog.Error(err)
 		return
 	}
 	for _, item := range listDR.Items {
-		if !util.Contains(listDisks, item.ObjectMeta.Name) {
+		if !util.Contains(devices, item.ObjectMeta.Name) {
```

Comment:
disk and device confuse me here.
if devices does not have a device then deactivate disk.
should we change `DeactivateDisk`
cc @akhilerm .

Reply:
@imazik  The following should fix the ambiguity.
```
if !util.Contains(disks, item.ObjectMeta.Name) {
```

> should we change `DeactivateDisk`

 You meant `DeactivateStaleDiskResource` ?